metricflow-semantics/metricflow_semantics/query/query_resolver.py

```diff
@@ -196,6 +197,18 @@ def _resolve_metric_inputs(
                     )
                 )
             else:
+                # Add the desired alias to the spec if it's provided in the input object
+                if (
+                    isinstance(metric_input.input_obj, MetricQueryParameter)
```

Comment:
So, we have generally tried to avoid putting any `isinstance` checks in MF. This is mostly a Paul thing, he tends to think they indicate a problem with the classes themselves if we need to add a check like this. Instead we would want to do something here that's similar to how `GroupByParamaters` are handled. It's a bit more complicated (I'm reading the code and haven't totally figured out the solution yet). I definitely don't think it's worth you trying to figure this out before you leave for vacation so I can pick up this PR from you!